tests/unit/test_snap.py

```diff
@@ -902,28 +902,24 @@ def fake_snap(command: str, optargs: Optional[Iterable[str]] = None) -> str:
         with self.assertRaises(TypeError):
             foo.get(None)  # pyright: ignore[reportArgumentType]
 
-    @patch("charms.operator_libs_linux.v2.snap.subprocess.check_output")
-    def test_snap_set_typed(self, mock_subprocess):
+    @patch("charms.operator_libs_linux.v2.snap.SnapClient.put_snap_config")
```

Comment:
Let's also add a test or two that tests the logic in the new `_wait` method.

Reply:
Added, thanks!